sst/sstable.go

```diff
@@ -147,32 +150,42 @@ func BuildSSTable(format common.DataFormat, buffSizeEstimate int, entriesEstimat
 	if metadataOffset > math.MaxUint32 {
 		return nil, nil, nil, 0, 0, errwrap.New("SSTable too big")
 	}
+
 	buff[1] = byte(metadataOffset)
 	buff[2] = byte(metadataOffset >> 8)
 	buff[3] = byte(metadataOffset >> 16)
 	buff[4] = byte(metadataOffset >> 24)
 
-	return &SSTable{
+	selfTable := &SSTable{
 		format:           format,
 		maxKeyLength:     uint32(maxKeyLength),
 		numEntries:       uint32(numEntries),
 		numDeletes:       uint32(numDeletes),
 		numPrefixDeletes: uint32(numPrefixDeletes),
 		indexOffset:      uint32(indexOffset),
 		creationTime:     uint64(time.Now().UTC().UnixMilli()),
-		data:             buff,
-	}, smallestKey, largestKey, minVersion, maxVersion, nil
+	}
+
+	buff, selfTable.metadataSize = selfTable.serializedMetdata(buff)
+	selfTable.data = buff
+
+	return selfTable, smallestKey, largestKey, minVersion, maxVersion, nil
+}
+
+func (s *SSTable) serializedMetdata(buff []byte) ([]byte, int) {
```

Comment:
Looking much better. As a further simplification there's no need to have a separate method for this, just do the appends in the BuildSSTable method

Reply:
I just named the function. Can do there as well.

Reply:
Addressed